sdks/java/core/src/test/java/org/apache/beam/sdk/io/TFRecordSchemaTransformProviderTest.java

```diff
@@ -0,0 +1,605 @@
+              .setCompression(compression)
+              .setFilePattern(filePattern)
+              .build()
+              .validate();
+        });
+
+    // Compression unset
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          TFRecordReadSchemaTransformConfiguration.builder()
+              .setValidate(false)
+              // .setCompression(Compression.AUTO) // Compression is mandatory
+              .setFilePattern(filePattern)
+              .build()
+              .validate();
+        });
+  }
+
+  @Test
+  public void testWriteInvalidConfigurations() throws Exception {
+    String fileName = "foo";
+    String nonExistentPath = "abc";
+    String filenameSuffix = "bar";
+    String shardTemplate = "xyz";
+    String compression = "AUTO";
+    Integer numShards = 10;
+
+    // Invalid outputPrefix
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          TFRecordWriteSchemaTransformConfiguration.builder()
+              .setOutputPrefix(tempFolder.getRoot().toPath().toString() + nonExistentPath)
+              .setFilenameSuffix(filenameSuffix)
+              .setShardTemplate(shardTemplate)
+              .setNumShards(numShards)
+              .setCompression(compression)
+              .setNoSpilling(true)
+              .build()
+              .validate();
+        });
+
+    // NumShards unset
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          TFRecordWriteSchemaTransformConfiguration.builder()
+              .setOutputPrefix(tempFolder.getRoot().toPath().toString() + fileName)
+              .setFilenameSuffix(filenameSuffix)
+              .setShardTemplate(shardTemplate)
+              // .setNumShards(numShards) // NumShards is mandatory
+              .setCompression(compression)
+              .setNoSpilling(true)
+              .build()
+              .validate();
+        });
+
+    // Compression unset
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          TFRecordWriteSchemaTransformConfiguration.builder()
+              .setOutputPrefix(tempFolder.getRoot().toPath().toString() + fileName)
+              .setFilenameSuffix(filenameSuffix)
+              .setShardTemplate(shardTemplate)
+              .setNumShards(numShards)
+              // .setCompression(compression) // Compression is mandatory
+              .setNoSpilling(true)
+              .build()
+              .validate();
+        });
+
+    // NoSpilling unset
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          TFRecordWriteSchemaTransformConfiguration.builder()
+              .setOutputPrefix(tempFolder.getRoot().toPath().toString() + fileName)
+              .setFilenameSuffix(filenameSuffix)
+              .setShardTemplate(shardTemplate)
+              .setNumShards(numShards)
+              .setCompression(compression)
+              // .setNoSpilling(true) // NoSpilling is mandatory
```

Comment:
Shouldn't this no longer throw? I'm a bit confused how this is still passing.

Reply:
Validate was also throwing an error, which allowed it to pass.  Will update.  Thanks.